apps/managedidentity/managedidentity.go

```diff
@@ -76,59 +79,154 @@ func (s systemAssignedValue) value() string { return string(s) }
-	return withHTTPClientOption{HttpClient: httpClient}
+func WithHTTPClient(httpClient ops.HTTPClient) ClientOption {
+	return func(o *ClientOptions) {
+		o.httpClient = httpClient
+	}
 }
 
 // Client to be used to acquire tokens for managed identity.
 // ID: [SystemAssigned()], [ClientID("clientID")], [ResourceID("resourceID")], [ObjectID("objectID")]
 //
 // Options: [WithHTTPClient]
-func New(id ID, options ...Option) (Client, error) {
+func New(id ID, options ...ClientOption) (Client, error) {
 	fmt.Println("idType: ", id.value())
 
-	opts := clientOptions{
-		claims: "claims",
+	opts := ClientOptions{ // work on this side where
+		httpClient: shared.DefaultClient,
 	}
 
 	for _, option := range options {
-		option.apply(&opts)
+		option(&opts)
 	}
 
-	authInfo, err := authority.NewInfoFromAuthorityURI("authorityURI", true, false)
+	client := Client{ // TODO :: check for http client
+		MiType:     id,
+		httpClient: opts.httpClient,
+	}
+
+	return client, nil
+}
+
+func getTokenForURL(url *url.URL, httpClient ops.HTTPClient) (accesstokens.TokenResponse, error) {
+	req, err := http.NewRequest(http.MethodGet, url.String(), nil)
 	if err != nil {
-		return Client{}, err
+		return accesstokens.TokenResponse{}, err
 	}
+	req.Header.Add("Metadata", "true")
 
-	authParams := authority.NewAuthParams(id.value(), authInfo)
-	client := Client{ // Note: Hey, don't even THINK about making Base into *Base. See "design notes" in public.go and confidential.go
-		AuthParams:      authParams,
-		cacheAccessorMu: &sync.RWMutex{},
-		// manager:         storage.New(token),
-		// pmanager:        storage.NewPartitionedManager(token),
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return accesstokens.TokenResponse{}, err
 	}
 
-	return client, err
+	// Pull out response body
+	responseBytes, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return accesstokens.TokenResponse{}, err
+	}
+
+	// Unmarshall response body into struct
+	var r accesstokens.TokenResponse
+	err = json.Unmarshal(responseBytes, &r)
+	if err != nil {
+		return accesstokens.TokenResponse{}, err
+	}
+	return r, nil
+
 }
 
 // Acquires tokens from the configured managed identity on an azure resource.
 //
 // Resource: scopes application is requesting access to
 // Options: [WithClaims]
 func (client Client) AcquireToken(context context.Context, resource string, options ...AcquireTokenOption) (base.AuthResult, error) {
-	return base.AuthResult{}, nil
+	o := AcquireTokenOptions{}
+
+	for _, option := range options {
+		option(&o)
+	}
+
+	// try and find some resource which can be accessed
+	// service fabric  GET
+	// app service  GET
+	// could shell  POST request
+	// azure arc  GET
+	// default :: IMDS  GET
+	//
+	// Sources that send GET requests: App Service, Azure Arc, IMDS, Service Fabric
+	//
+	// Sources that send POST requests: Cloud Shell
+
+	var msiEndpoint *url.URL
+	msiEndpoint, err := url.Parse(IMDSEndpoint)
```

Comment:
We don't need to parse this every time--it's constant

Reply:
We will need to parse the URL constant no ?
I could not find a way to make url constant as url is struct.

Reply:
Sorry for the distraction, you make a good point: although you don't strictly speaking need a `url.URL`, that type's API is the most convenient way to format a query string.